File: Largest Continuous Sum/Kadane.java
package com.example;

import static java.lang.Math.max;

public class Kadane {

	public static void main(String[] args) {

		int arr[] = { 1, -1, -1, 2, -2, 3 };
		int sum = largestSum(arr);
		System.out.println(sum);
	}

	public static int largestSum(int[] arr) {

		int maxSum = arr[0];
		int currentSum = arr[0];

		for (int i = 1; i < arr.length; i++) {
			currentSum = max(currentSum + arr[i], arr[i]);
			maxSum = max(maxSum, currentSum);
		}

		return maxSum;
	}

}

File: Largest Continuous Sum/kadane.py
def largest_cont_sum(arr):
    current_sum = max_sum = arr[0]

    for num in arr[1:]:
        current_sum = max(current_sum + num, num)
        max_sum = max(max_sum, current_sum)

    return max_sum;


a = [1, -1, -1, 2, -2, 3]
print largest_cont_sum(a)